### PR TITLE
Dossier templates can reference document templates.

### PR DESCRIPTION
**Pull request by elioschmutz**
This PR extends the dossier template schema with a `related_documents` field. It allows document templates as references which will be created as new documents when adding a new dossier from a dossier template.

For [CA-4519]

## Checklist

_Everything has to be done/checked. Checked but not present means the author deemed it unnecessary._

- [x] Changelog entry
- [x] Link to issue (Jira or GitHub) and backlink in issue (Jira)

_Only applicable should be left and checked._

- Upgrade-Steps:
  - [ ] SQL Operations do not use imported model (see [docs](https://4teamwork.atlassian.net/wiki/spaces/4TEAM/pages/994344994/Upgrade-Steps))
  - [ ] Make it deferrable if possible
  - [ ] Execute as much as possible conditionally
  - DB-Schema migration
    - [ ] All changes on a model (columns, etc) are included in a DB-schema migration.
    - [ ] Constraint names are shorter than 30 characters (`Oracle`)
- API change:
  - [ ] Documentation is updated
  - [ ] API Changelog entry (see [guide](https://4teamwork.atlassian.net/wiki/spaces/4TEAM/pages/451248812/API+Changelog+Guidelines))
  - If breaking:
    - [ ] api-change label added
    - [ ] #delivery channel notified about breaking change
    - [ ] Scrum master is informed
- Bug fixed:
  - [ ] Resolved any Sentry issues caused by this bug
- New functionality:
  - [ ] for `document` also works for `mail`
  - [ ] for `task` also works for `forwarding`
- Further improvements needed:
  - [ ] Create follow-up stories and link them in the PR and Jira issue
- [ ] Change could impact client installations, client policies need to be adapted
- New translations
  - [ ] All msg-strings are unicode
- Change in schema definition:
  - [ ] If `missing_value` is specified, then `default` has to be set to the same value

[CA-4517]: https://4teamwork.atlassian.net/browse/CA-4517?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[CA-4519]: https://4teamwork.atlassian.net/browse/CA-4519?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by njohner — COMMENTED** [review]
See comment.
Also this story is blocked by https://4teamwork.atlassian.net/browse/CA-5463, for which I have an open PR that still needs a review. Once mine is merged you'll need to add an indexer for your new field.

File: opengever/dossier/dossiertemplate/behaviors.py [hunk 1]
```diff
@@ -111,6 +112,24 @@ class IDossierTemplate(model.Schema):
...
+        default=list(),
+        missing_value=list(),
+        value_type=RelationChoice(
+            title=u'Related document templates',
+            source=SolrObjPathSourceBinder(
+                portal_type=("opengever.document.document", ),
+                navigation_tree_query={
+                    'object_provides':
+                    ['opengever.dossier.templatefolder.interfaces.ITemplateFolder',
+                     'opengever.document.document.IDocumentSchema']
```
[the remark this answers is not in the document]
Comment (elioschmutz) [reply opengever/dossier/dossiertemplate/behaviors.py:127]: We do not have such an interface... That's the old known issue of document templates that we can only differ between default documents by its parent (templatefolder).

**Comment by elioschmutz** [comment]
@njohner I added the indexer in a new commit.

File: opengever/dossier/dossiertemplate/behaviors.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (njohner) [reply opengever/dossier/dossiertemplate/behaviors.py:127]: Sorry, `ITemplateDocumentMarker`

**Comment by elioschmutz** [comment]
@njohner I updated the PR. Could you take another look at it?

File: opengever/api/listing_stats.py [hunk 2]
```diff
@@ -19,7 +20,7 @@ def get_path_depth(context):
 
 @implementer(IExpandableElement)
 @adapter(Interface, IOpengeverBaseLayer)
-class ListingStats(object):
+class ListingStats(object, RequestPayloadMixin):
```
Comment (njohner) [inline opengever/api/listing_stats.py:23]: I guess that line should be in the previous commit

**Pull request merged**